spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala

```diff
@@ -0,0 +1,620 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.serializer
+
+import scala.collection.JavaConverters._
+import scala.reflect.runtime.universe
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.optim.{L1L2Regularizer, L1Regularizer, L2Regularizer, Regularizer}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import serialization.Model._
+import serialization.Model.AttrValue.{ArrayValue, DataType}
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+
+/**
+ * Trait which defines get attribute value from saved protobuf data and convert BigDL object to
+ * protobuf format attribute data
+ */
+trait DataConverter {
+
+/**
+ * Get attribute value from protobuf attribute data
+ * @tparam T data type
+ * @param attribute  protobuf generated Attribute instance
+ * @return BigDL compatible param value
+ */
+  def getAttributeValue[T : ClassTag](attribute: AttrValue)(
+    implicit ev: TensorNumeric[T]) : AnyRef
+
+/**
+ * Set attribute value to protobuf format
+ * @tparam T data type
+ * @param attributeBuilder  the attribute value writable instance
+ * @param value the value to be written to protobuf file
+ * @param valueType the type of the value to help set the data type
+ */
+  def setAttributeValue[T : ClassTag](attributeBuilder : AttrValue.Builder, value: Any,
```

Comment:
it' used in sub class (or specifically in sub object which use the TensorNumeric to infer the type or transfer data)